### PR TITLE
(#1478) Error State Color

### LINKED ISSUE #1478: Story: Add Error State Modifier for Dark Backgrounds for Form Inputs
### New Feature Request
This request is to add a modifier to the Error State for File Inputs so that it meets accessibility requirements when displayed on dark backgrounds.

Classes that have `error-dark`:
**Components**
* `.usa-input--error`
* `.usa-form-group--error`
* `.usa-input-group--error`
   * Located in `usa-input-prefix-suffix`
* `* .usa-form-group--error .usa-file-input__target` 
    * we manually patched to `error-dark`
*  `.has-invalid-file .usa-file-input__accepted-files-message` 
     * we manually patched to `error-dark`
*  `.usa-character-count__status.usa-character-count__status--invalid`
    * Wins for stupidest class name
* `.usa-label--required`
   * Which is kind of confusing because `usa-label--error` is only bolding the text. 
* `.usa-error-message`
* .has-invalid-file .usa-file-input__accepted-files-message
   * this is actually in our patch.

### What's the expected change?
- A modifier will be made to the Error State so that it will meet accessibility requirements on dark backgrounds.
- All error states need to change from `error-dark` to `error`
- Use the [comp for the Error State Modifier](https://www.figma.com/file/3LA0fAGyGdzpBeNDCoulFf/NCIDS-Design-Kit---Master?type=design&node-id=6374-11972&mode=design)

   
### What's the current functionality?
- There is no modifier of the error state for use on dark backgrounds.
- Error states use `error-dark`

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #1478

### CODE REVIEW

**Comment by blilianyu** [comment]
On Form Elements, can we:
- make the background on the error message span the width of the input field
- add padding on the text so it's not touching
- add the background so it's touching/around the input field?

Screenshot 2024-02-07 at 10 53 26 AM

**Comment by karlikpj** [comment]
@blilianyu We can add padding  - however the error message itself is a separate element from the input (which could be a select / text / radio) and there is no consistent container that wraps them both in the HTML that would they red background to encompass the message and form control.
Screenshot 2024-02-07 at 11 08 54 AM

**Comment by blilianyu** [comment]
@karlikpj ok, then can we remove the margin so the error message is touching the input field?

**Comment by blilianyu** [comment]
@karlikpj updates look good! I would like to change the input field width to be 100% but noting that this change would be a separate ticket.

**Comment by blilianyu** [comment]
Updates on form field and the other error states look good! I've opened a CR #1520 for updating the input field width.

**Comment by arcepaul** [comment]
Looks like there is some right padding needed in the box
error-box

**Comment by blilianyu** [comment]
Updates look good to me!

File: testing/ncids-css-testing/stories/components/forms/form-elements/form-elements.stories.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { TestCase } from '../../../../components/test-case';
+import css from './index.scss';
+
+export default {
+	title: 'Components/Forms/Form Elements',
+};
+
+const Template = (args) => Component(args);
+
+export const Elements = () => <TestCase css={css} html={Template.bind({})()} />;
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/forms/form-elements/form-elements.stories.js:12]: Can you make sure this story matches the outline Bryan provided in the comment here
https://github.com/NCIOCPL/ncids/issues/1478#issuecomment-1922156326

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — APPROVED** [review]

**Review by olitharp-nci — APPROVED** [review]

**Review by sarinapadilla — APPROVED** [review]
Looks good!

**Pull request merged**